### DISCUSSION TITLE
Got OOM, how can I debug? (A param to get linters detail)

### DISCUSSION THREAD

**Discussion by jinleileiking** [golangci/golangci-lint#4058] [Q&A] [answered]
I only can see golangci-lint will got alot of rss, no linters detail dumped.

```
❯ golangci-lint run --print-resources-usage   -v
INFO [config_reader] Config search paths: [./ /home/jinlei/gocodes/src/cloudlive /home/jinlei/gocodes/src /home/jinlei/gocodes /home/jinlei /home /]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 7 linters: [errcheck gosimple govet ineffassign staticcheck typecheck unused]
INFO [loader] Go packages loading at mode 575 (deps|files|compiled_files|imports|name|types_sizes|exports_file) took 493.067629ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 9.362339ms

[1]    3503115 killed     golangci-lint run --print-resources-usage -v

```

**Comment by ldez** [comment, marked as answer]
1. Use the latest version of golangci-lint for your OS https://github.com/golangci/golangci-lint/releases
2. Use go1.20 or go1.21 https://go.dev/dl/
3. Be sure that your project compiles
4. Disable the linters one by one and see what happens

**Reply by jinleileiking** [reply]
Disable the linters one by one and see what happens
--- This sucks, cost a lot of time :(

**Reply by bombsimon** [reply]
Who do you think should do this for you instead? Using `golangci-lint` which is completely free is optional, you don't have to use it.

Also not that time consuming:

```sh
for linter in errcheck gosimple govet ineffassign staticcheck typecheck unused; \
    do golangci-lint run --disable-all --enable $linter --print-resources-usage -v; \
done
```

**Reply by jinleileiking** [reply]
I just want somebody help, I do not ask any body to do this...
